python/tskit/trees.py

```diff
@@ -2242,6 +2242,22 @@ def num_migrations(self):
         """
         return self._ll_tree_sequence.get_num_migrations()
 
+    @property
+    def oldest_time(self):
+        """
+        Returns the oldest time for any of the edges in this tree sequence.
```

Comment:
The name isn't quite right I think, as you might expect this to be max(tables.nodes.time) instead. What about ``max_parent_time``? "Returns the time of the oldest node that is a parent in an edge" (or something). Raises ValueError if there are no edges.

Probably simpler to define ``max_time`` to be honest. In practise the time needed to run np.max(tables.nodes.time) is going to be negligible (particularly for anything that small enough to draw, which is the application, right)?

Reply:
I see this as the oldest time in the *tree sequence*, by which I mean the sequence of trees, not the node table. I don't think of this necessarily the same as the oldest node.

Reply:
I just asked Wilder what he thought. He agreed that he thought of the oldest time in a tree sequence as the oldest point in any tree in the ts. That might be distinct from the oldest node time, which might be better accessed through the property `oldest_node`.

Either way, if we are drawing trees, we want the oldest point in any of the trees, not (necessarily) the oldest node in the node table, although they are likely to be the same aside from pathological cases.

Reply:
A tree sequence is defined as the tables, so I would expect ``oldest_time`` to be the maximum node time. Fine to add another property, but it must have a descriptve name that is clearly distinct from max_node_time.

Reply:
Fair enough. I guess since we can have unary / recombination nodes, it's not right to use e.g. `oldest_coalescence_time` or `max_tmrca`. Possibilities include `max_tree_depth`, `max_tree_time`. Or other suggestions (ping @awohns )

Reply:
Actually, I quite like `oldest_root`, or perhaps (to make it clear that we aren't returning the root itself, but the time of the oldest root, `oldest_root_time`.

Reply:
``oldest_root_time`` is good, but we would need to make sure we defined it as ``max(ts.node(u) for u in ts.samples())`` when there are zero edges (because of our definition of roots).